python/python/lance/fragment.py

```diff
@@ -384,3 +385,76 @@ def metadata(self) -> FragmentMetadata:
         FragmentMetadata
         """
         return FragmentMetadata(self._fragment.metadata().json())
+
+
+def write_fragments(
+    data: ReaderLike,
+    dataset_uri: Union[str, Path],
+    schema: Optional[pa.Schema] = None,
+    *,
+    max_rows_per_file: int = 1024 * 1024,
+    max_rows_per_group: int = 1024,
+    max_bytes_per_file: int = 90 * 1024 * 1024 * 1024,
+    progress: Optional[FragmentWriteProgress] = None,
+) -> List[FragmentMetadata]:
+    """
+    Write data into one or more fragments.
+
+    .. warning::
+
+        This is a low-level API intended for manually implementing distributed
+        writes. For most users, :func:`lance.write_dataset` is the recommended API.
+
+    Parameters
+    ----------
+    data : pa.Table or pa.RecordBatchReader
+        The data to be written to the fragment.
+    dataset_uri : str
+        The URI of the dataset.
+    schema : pa.Schema, optional
+        The schema of the data. If not specified, the schema will be inferred
+        from the data.
+    max_rows_per_file : int, default 1024 * 1024
+        The maximum number of rows per data file.
+    max_rows_per_group : int, default 1024
+        The maximum number of rows per group in the data file.
+    max_bytes_per_file : int, default 90 * 1024 * 1024 * 1024
+        The maximum number of bytes per data file. Defaults to 90 GB. This is a
```

Comment:
Can you explain why it is 90GB?

Reply:
Sure. I can mention there is hard 100GB limit in object stores.